src/coreclr/vm/methoditer.cpp

```diff
@@ -203,15 +203,11 @@ MethodDesc *LoadedMethodDescIterator::Current()
         return m_mainMD;
     }
 
-    MethodTable *pMT = m_typeIteratorEntry->GetTypeHandle().GetMethodTable();
+    MethodTable *pMT = m_typeIteratorEntry->GetTypeHandle().GetMethodTable()->GetCanonicalMethodTable();
     PREFIX_ASSUME(pMT != NULL);
-    _ASSERTE(pMT);
-
-    return pMT->GetMethodDescForSlot(m_mainMD->GetSlot());
+    return pMT->GetParallelMethodDesc(m_mainMD);
```

Comment:
The code for `GetParallelMethodDesc()` handles the EnC case by performing a slow path (i.e., `GetParallelMethodDescForEnC()`) because the slot is invalid&mdash;see [here](https://github.com/dotnet/runtime/blob/4cb00eacbeb6b9e40da29b8e9273707f94bd4a6c/src/coreclr/vm/methodtable.cpp#L8760-L8814). Note that the original call in this file was the same as the non-EnC scenario in `GetParallelMethodDesc()`. I could have made the check here too, but instead I just called `GetParallelMethodDesc()` which handles both EnC and non-EnC scenarios.